tools/lib/capitalization.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import absolute_import
+
+from typing import List, Tuple
+import re
+
+from bs4 import BeautifulSoup
+
+DISALLOWED_REGEXES = [re.compile(regex) for regex in [
+    '^[a-z]',
+    '\.\s+[a-z]',
+    '!\s+[a-z]',
+    '\?\s+[a-z]',
+    '^[A-Z][a-z]+[\sa-z0-9]+[A-Z]',
+]]
+
+# Keep this list lexicographically sorted.
+IGNORED_PHRASES = [
+    "I'm",
+    'API',
+    'Dropbox',
+    'GitHub',
+    'Google',
+    'I say',
+    'I want',
+    'ID',
+    'JIRA',
+    'JSON',
+    'Kerberos',
+    'Mac',
+    'MiB',
+    'Pivotal',
+    'SSO',
+    'URL',
+    'Ubuntu',
+    'Webathena',
+    'Windows',
+    'WordPress',
+    'XML',
+    'Zephyr',
+    'Zulip',
+    'and',
+    'bot',
+    'images',
+    'left',
+]
+
+START_PLACEHOLDER = 'Ignored_phrase_at_the_start'
+PUNCTUATION_PLACEHOLDER = 'Ignored_phrase_after_punctuation'
+
+TRANSLATION_FUNCTION = [
+    ('^\s*{}', START_PLACEHOLDER),
+    ('\.\s+{}', '. ' + PUNCTUATION_PLACEHOLDER),
+    ('!\s+{}', '! ' + PUNCTUATION_PLACEHOLDER),
+    ('\?\s+{}', '? ' + PUNCTUATION_PLACEHOLDER),
+]
+
+def apply_translation_function(string):
+    # type: (str) -> str
+    for phrase in IGNORED_PHRASES:
+        for regex_template, placeholder in TRANSLATION_FUNCTION:
+            regex = regex_template.format(phrase)
+            string = re.sub(regex, placeholder, string)
+
+        string = string.replace(phrase, phrase.lower())
+        string = string.replace(START_PLACEHOLDER, phrase.capitalize())
```

Comment:
I would add comments here explaining the purposes of the PLACHEHOLDERs.  For the punctuation piece, are sure it doesn't make sense to do something a little less hacky like break the string into a list of phrases?